wallet-new/src/Cardano/Wallet/Kernel/DB/TxMeta/Types.hs

```diff
@@ -186,10 +185,16 @@ instance Buildable TxMetaStorageError where
 -- | Generates 'NonEmpty' collections which do not contain duplicates.
 -- Limit the size to @size@ elements.
 uniqueElements :: (Arbitrary a, Ord a) => Int -> Gen (NonEmpty a)
+uniqueElements 0 = error "uniqueElements should have size > 0"
 uniqueElements size = do
-    noDupes <- suchThat arbitrary (\s -> length s >= size)
-    let (e, es) = Set.deleteFindMax noDupes
-    return $ e :| List.take size (Set.toList es)
+    hd <- arbitrary
+    (\st -> hd :| (Set.toList st)) <$> (go mempty 0)
```

Comment:
Perhaps a silly question -- if I understand the algorithm, we call `go` and we stop recursing after we collected enough elements. However, when we cons `hd` to the created set, what guarantees that `hd` is unique with respect to all the other elements?

Reply:
no silly questions, just silly bugs